### ISSUE TITLE
Updating device provisioning for new attributes may not be working.

### ISSUE THREAD

**Issue by ezequiel-umu** [telefonicaid/iotagent-node-lib#277]
Using this curl script for device providing:

```
#!/bin/bash

(curl localhost:4041/iot/devices -s -S --header 'Content-Type: application/json' \
  --header 'Accept: application/json' --header 'fiware-service: ' --header 'fiware-servicepath: /mqtt' \
  -d @- | python -mjson.tool) <<EOF
{
  "devices": [
    {
      "device_id": "MQTT",
      "protocol": "GENERIC_PROTO",
      "entity_name": "MQTT Device",
      "entity_type": "Device",
      "timezone": "America/Santiago"
    }
  ]
}
EOF
```

And then this for device updating:

```
#!/bin/bash

(curl localhost:4041/iot/devices/MQTT -s -S -X PUT --header 'Content-Type: application/json' \
  --header 'Accept: application/json' --header 'fiware-service: ' --header 'fiware-servicepath: /mqtt' \
  -d @- | python -mjson.tool) <<EOF
{
  "attributes": [
    {
        "name": "nuevoAtributo",
        "type": "Integer"
    }
  ]
}
EOF
```

With MQTT IoTAgent (which uses this library) triggers this:

```
time=2016-02-16T09:36:00.090Z | lvl=DEBUG | corr=n/a | trans=n/a | op=IoTAgentNGSI.NorthboundServer | msg=Body:

{
    "attributes": [
        {
            "name": "nuevoAtributo",
            "type": "Integer"
        }
    ]
}


time=2016-02-16T09:36:00.093Z | lvl=DEBUG | corr=n/a | trans=n/a | op=IoTAgentNGSI.NorthboundServer | msg=Error [TypeError] handing request: Cannot read property '0' of undefined
```

Any idea?

**Comment by dmoranj** [comment]
Seems like a bug to me. Maybe there is some piece of information that should be set mandatory (I'm thinking about the type, in particular). Try adding the entity_type field to your request. 

In any case, if it was a problem concerning mandatory and optional types, we should improve the payload checking. I'll label it as bug and check it.

**Comment by ezequiel-umu** [comment]
With:

```
#!/bin/bash

(curl localhost:4041/iot/devices/MQTT -s -S -X PUT --header 'Content-Type: application/json' \
  --header 'Accept: application/json' --header 'fiware-service: odins' --header 'fiware-servicepath: /mqtt' \
  -d @- | python -mjson.tool) <<EOF
{
  "entity_type": "Device",
  "attributes": [
    {
        "name": "nuevoAtributo",
        "type": "Integer"
    }
  ]
}
EOF
```

The IoTAMQTT shows:

```
time=2016-02-16T11:05:25.175Z | lvl=DEBUG | corr=n/a | trans=n/a | op=IoTAgentNGSI.NorthboundServer | msg=Body:

{
    "entity_type": "Device",
    "attributes": [
        {
            "name": "nuevoAtributo",
            "type": "Integer"
        }
    ]
}


time=2016-02-16T11:05:25.177Z | lvl=DEBUG | corr=n/a | trans=n/a | op=IoTAgentNGSI.RestUtils | msg=Errors found validating request: {"valid":false,"errors":[{"attribute":"additionalProperties","property":"entity_type","actual":"Device","message":"must not exist"}]}
time=2016-02-16T11:05:25.178Z | lvl=DEBUG | corr=n/a | trans=n/a | op=IoTAgentNGSI.NorthboundServer | msg=Error [WRONG_SYNTAX] handing request: Wrong syntax in request: Errors found validating request.
```

**Comment by dmoranj** [comment]
Fixed in #280
